ocs_ci/ocs/ocs_upgrade.py

```diff
@@ -385,3 +390,112 @@ def set_upgrade_images(self):
+        log.debug(f"ocs operator pod info: {ocs_operator_pod}")
+        ocs_operator_node = get_pod_node(ocs_operator_pod)
+
+        return get_node_name(ocs_operator_node)
+
+    def worker_node_shutdown(self, nodes, abrupt):
+        """
+        Shutdown worker node that running ocs-operator pod
+
+        Args:
+            abrupt: bool: True if abrupt shutdown, False for permanent shutdown
+
+        """
+        log.info(f"Abrupt {abrupt}")
+        # get ocs-operator node:
+        ocs_operator_node_name = self.get_ocs_operator_node_name()
+
+        # get workers node objects:
+        node_to_shutdown = list()
+        for node in get_typed_nodes():
+            node_name = get_node_name(node)
+            log.info(f"node: {node_name}, ocs operator node: {ocs_operator_node_name}")
+            if node_name == ocs_operator_node_name:
+                # if node_name != rook_ct_node_name:
+                node_to_shutdown.append(node)
+                log.info(f"node to shutdown: {get_node_name(node_to_shutdown[0])}")
+                nodes.stop_nodes(node_to_shutdown)
+                log.info("stop instance - done!")
+                break
+
+        log.info("Sleeping 5 minutes")
+        time.sleep(320)
+        assert wait_for_ct_pod_recovery(), "Ceph tools pod failed to come up on another node"
+        if abrupt:
+            log.info("Abrupt Shutdown")
+            if node_to_shutdown:
+                nodes.start_nodes(nodes=node_to_shutdown)
+
+
+@pytest.mark.parametrize()
+def run_ocs_upgrade(nodes, operation=None, operation_args=None):
+    """
+    Tests upgrade procedure of OCS cluster
+
+    """
+    ceph_cluster = CephCluster()
+    upgrade_ocs = OCSUpgrade(
+        namespace=config.ENV_DATA['cluster_namespace'],
+        version_before_upgrade=config.ENV_DATA.get("ocs_version"),
+        ocs_registry_image=config.UPGRADE.get('upgrade_ocs_registry_image'),
+        upgrade_in_current_source=config.UPGRADE.get('upgrade_in_current_source', False)
+    )
+    upgrade_version = upgrade_ocs.get_upgrade_version()
+    assert upgrade_ocs.get_parsed_versions()[1] >= upgrade_ocs.get_parsed_versions()[0], (
+        f"Version you would like to upgrade to: {upgrade_version} "
+        f"is not higher or equal to the version you currently running: "
+        f"{upgrade_ocs.version_before_upgrade}"
+    )
+    csv_name_pre_upgrade = upgrade_ocs.get_csv_name_pre_upgrade()
+    pre_upgrade_images = upgrade_ocs.get_pre_upgrade_image(csv_name_pre_upgrade)
+    upgrade_ocs.load_version_config_file(upgrade_version)
+    with CephHealthMonitor(ceph_cluster):
+        channel = upgrade_ocs.set_upgrade_channel()
+        upgrade_ocs.set_upgrade_images()
+        upgrade_ocs.update_subscription(channel)
+        if operation and operation_args:
+            log.info("**** tier4a test starting here ****")
+            log.info(f"running test function: {operation}")
+            _ = operation(nodes, operation_args)
```

Comment:
The call of operation should be done more generic to accept args and kwargs.  And in this call you should do something like 
```python
operation(*operation_args, **operation_kwargs)
```
Where `operation_args` will be list of all arguments and `operation_kwargs` will be dictionary of named arguments.

BTW why we need to pass nodes here? It will basically require each operation needs to take this nodes as first parameter where we close the door to have all this more general and reusable for other operations.